src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Hosting;
+    using Logging;
+
+    class StartableAndStoppableRunner
+    {
+        public StartableAndStoppableRunner(IEnumerable<IWantToRunWhenEndpointStartsAndStops> wantToRunWhenBusStartsAndStops)
+        {
+            this.wantToRunWhenBusStartsAndStops = wantToRunWhenBusStartsAndStops;
+        }
+
+        public Task Start(IMessageSession session)
+        {
+            var startableTasks = new List<Task>();
+            foreach (var startable in wantToRunWhenBusStartsAndStops)
+            {
+                var task = startable.Start(session).ThrowIfNull();
+
+                var startable1 = startable;
+                task.ContinueWith(t =>
+                {
+                    thingsRanAtStartup.Add(startable1);
+                    Log.DebugFormat("Started {0}.", startable1.GetType().AssemblyQualifiedName);
+                }, TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously);
+                task.ContinueWith(t => { Log.Error($"Startup task {startable1.GetType().AssemblyQualifiedName} failed to complete.", t.Exception); }, TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously);
+
+                startableTasks.Add(task);
```

Comment:
Done by me when I ported the original code. It was primarily done because of the conditional logging. Since we want to start them concurrently and log per instance there is not much we can do else. I initially asked if we could remove the logs. The answer was no.